src/Endpoint/EndpointBase.php

```diff
@@ -263,11 +281,19 @@ protected function getRequestBody(): ?StreamInterface
             return $this->streamFactory->createStream(http_build_query($parts));
         }
 
+        // Simple JSON body.
+        if ($json = $this->getRequestJsonBody()) {
```

Comment:
This case was not covered by the original Open Europa Search client. We simply need to send a JSON body.